src/postprocess/band_structure.jl

```diff
@@ -19,18 +19,21 @@ function high_symmetry_kpath(model; kline_density=20)
     # https://github.com/louisponet/DFControl.jl/blob/master/src/structure.jl
     # but for this the best way to go would be to refactor into a small "CrystalStructure"
     # julia module which deals with these sort of low-level details everyone can agree on.
-    pystructure = pymatgen_structure(model.lattice, model.atoms)
-    symm_kpath = pyimport("pymatgen.symmetry.bandstructure").HighSymmKpath(pystructure)
-    kcoords, labels = symm_kpath.get_kpoints(kline_density, coords_are_cartesian=false)
-
+    
+    sgnum = spglib_get_spacegroup(model.lattice, model.atoms)
+    Rs = [model.lattice[i, :] for i in 1:size(model.lattice,1)]
+    kp        = Brillouin.irrfbz_path(sgnum, Rs)
+    N         = kline_density*length(vcat(kp.paths...))
```

Comment:
I suspect this isn't jiving exactly with the differing notions of k-point density here: Brillouin's `interpolate` will try to distribute `N` points as equidistantly as possible across the entire path (with the distance metric evaluated in Cartesian space) - but `length(vcat(kp.paths))` will just give the number of unconnected paths overall. 
Maybe what you want a measure of the [overall path length](https://github.com/thchr/Brillouin.jl/blob/212eec65e97b3b9dcacfea9d02828b8c70ac6f55/src/interpolate-paths.jl#L102-L108).

Reply:
Yes, this would be much better, I think. Thanks!

Reply:
@harrisonlabollita Just to rule out another common pitfall: Have you taken unit conversions into account? Notice my comment about the `kline_density` in the 1D case for example: The kline density is in Angströms. I'm not sure what unit system is used for Brillouin. If I get it right from quickly looking at the code the carthesian coords in Brillouin are essential the units used by the lattice (right @thchr ?). So in that case you have a unit clash because our lattice is given in Bohrs. 

Also the kp is in fractional coords, so you need to make user to use cartesian lengths to compute the number of kpoints. 

OTOH it might make sense for Brillouin.jl to add a method, that takes a desired k-point density and does all the magic @thchr?

Reply:
Hmm, Brillouin doesn't make a concrete choice of any unit system: the impact of units should carry over through whatever you specify in the lattice vectors `Rs`: by default, `irrfbz_path` will return a **k**-path specified in reduced coordinates (i.e. in the basis of *primitive* reciprocal lattice). Similarly, if you pass that path to `interpolate`, the resulting interpolation points will be in reduced coordinates.
Internally, Brillouin's `interpolate` will first convert to cartesian coordinates to figure out the distances between segments in the **k**-path properly (just to be in an orthonormal basis) - but what it returns will be in reduced coordinates.

It might make sense for Brillouin to have an version of `interpolate` that takes a `density` kwarg - but I'm not completely sure about how to specify density meaningfully in the absence of a concrete unit system choice.

(The reason Brillouin doesn't have a default unit choice is that I'm in photonics where the units end up cancelling out due to a frequency-scale invariance)

Reply:
While I remember it: one other possible convention pitfall is that Brillouin's `irrfbz_path` requires that you specify the direct (i.e. real-space) basis in a *conventional* setting (e.g., for, say, diamond this would be a cubic lattice) - but returns the reciprocal path in the basis of the *primitive* reciprocal basis.

This is just to say that it is important that `Rs` refers to the conventional direct basis (even though the unit cell you'd use in your calculation would be in the primitive direct basis). This was basically just adopted from the convention that SeeK-path went with. I don't know which setting spglib returns the basis in - so I just wanted to point this out clearly. 

(Right now, Brillouin doesn't have methods to transform back and forth between conventional/primitive settings for direct bases, only for reciprocal bases - but if you need that, I can add it)

Reply:
Yes this could be an issue @thchr Thanks for pointing that out. I'll do some research about the defails.